### DISCUSSION TITLE
Post::model() - quick instantiation

### DISCUSSION THREAD

**Discussion by marknuyens** [laravel/framework#48794] [Ideas]
Hello everyone,

I would like to contribute a potential solution for quickly constructing a class using a new static `model()` method. Straight-forward, but has proven to be useful, especially for "method chaining".

It's actually similar to the static `make()` method for Json Resources.

```php
/**
 * Create a new Eloquent model instance.
 *
 * @param  array  $attributes
 * @return Model
 */
public static function model(array $attributes = []): Model
{
    return new static($attributes);
}
```
I've been using this utility function myself and it has served me well in a couple of scenarios, mostly whenever I just wanted the model's primary key or other kind of properties that were not related to its data.

Although mine originally didn't support the `$attributes` parameter, I would imagine this could be beneficial too.

Anyway, would love to to hear your thoughts!

**Comment by shaedrich** [comment]
So, what's the difference to `(new Model($attributes))->otherMethod();` resp. `Model::create($attributes)->otherMethod()`?

**Reply by marknuyens** [reply]
I think my method is a lot cleaner as opposed to `(new Model)->otherMethod()`. And it’s not like the `create()` method in that it doesn’t create or store anything. It just instantiates a new model class, optionally with attributes.

**Reply by shaedrich** [reply]
Okay, so it doesn't add any value other than being one character longer but a little more convenient. Even though, that might be nice, I don't think, Taylor Otwell will add it.

**Reply by marknuyens** [reply]
If Taylor would add it, it would make sense ;) ;)

**Reply by shaedrich** [reply]
I admire your hopes, however, they are usually not well-invested with Taylor

**Reply by marknuyens** [reply]
All jokes aside, the idea is easy to implement, so perhaps people will just attach this method to their own base model. It’s really just a way to improve DX, rather than bringing anything new or sparing keystrokes.

**Reply by shaedrich** [reply]
In all seriousness: It won't convince Taylor

**Reply by macropay-solutions** [reply]
@shaedrich

 > So, what's the difference to `(new Model($attributes))->otherMethod();` resp. `Model::create($attributes)->otherMethod()`?

The left example does not save the model, just instantiate and fills it. The right instantiates it, fills it and also saves it in db, returning the new model with primary key generated if any.
Anyway I agree with you that the method is just an overhead.

**Reply by shaedrich** [reply]
> The left example does not save the model, just instantiate and fills it. The right instantiates it, fills it and also saves it in db, returning the new model with primary key generated if any.

I wasn't comparing `new` to `create()` 😅 I was comparing them to the proposed method 😉
